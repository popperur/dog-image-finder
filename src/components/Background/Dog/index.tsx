import {
  BackLeg,
  BackLegFoot,
  DogBackTorso,
  DogContainer,
  FrontLeftFoot,
  FrontLeftLeg,
  FrontRightFoot,
  FrontRightLeg,
} from "components/Background/Dog/styles.tsx";
import DogHead from "components/Background/Dog/DogHead.tsx";
import DogTail from "components/Background/Dog/DogTail.tsx";

function Dog() {
  return (
    <DogContainer>
      <FrontLeftLeg>
        <FrontLeftFoot />
      </FrontLeftLeg>
      <DogBackTorso />
      <DogHead />
      <DogTail />
      <BackLeg>
        <BackLegFoot />
      </BackLeg>
      <FrontRightLeg>
        <FrontRightFoot />
      </FrontRightLeg>
    </DogContainer>
  );
}

export default Dog;
