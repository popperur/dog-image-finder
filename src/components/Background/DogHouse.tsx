import {
  DogHouseContainer,
  RoofLeft,
  RoofRight,
  RoofRightWall,
  WallLeft,
  WallRight,
  WallRightDoor,
} from "components/Background/styles.tsx";

function DogHouse() {
  return (
    <DogHouseContainer>
      <RoofRight>
        <RoofRightWall />
      </RoofRight>
      <WallLeft />
      <WallRight>
        <WallRightDoor />
      </WallRight>
      <RoofLeft />
    </DogHouseContainer>
  );
}

export default DogHouse;
