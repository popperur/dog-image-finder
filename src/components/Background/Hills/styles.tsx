import styled from "styled-components";

export const HillContainer = styled.div``;

export const HillHolder = styled.div<{ $styles?: string[] }>`
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  ${props => props.$styles && props.$styles.join(";")};
`;

export const BackHill = styled.div`
  position: absolute;
  bottom: 0;
  right: 5%;
  height: 250px;
  width: 400px;
  background-color: #1f2731;
  border-top-left-radius: 500px;
  border-top-right-radius: 500px;
`;
