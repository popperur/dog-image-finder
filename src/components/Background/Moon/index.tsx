import {
  MoonCenter,
  MoonContainer,
} from "components/Background/Moon/styles.tsx";
import MoonCraters from "components/Background/Moon/MoonCraters";
import MoonGlows from "components/Background/Moon/MoonGlows";

function Moon() {
  return (
    <MoonContainer>
      <MoonGlows />
      <MoonCenter>
        <MoonCraters />
      </MoonCenter>
    </MoonContainer>
  );
}

export default Moon;
