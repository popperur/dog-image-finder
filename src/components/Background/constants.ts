export const TOOLTIP_PAUSED =
  "Radio signal from the Moon! This incredible discovery could greatly enhance your experience in browsing for dogs!";
export const TOOLTIP_PLAYING = "Auuuuuu!";

// top, left, width, height, animationDelay
export const STARS = [
  [45, 66, 6, 6, 2],
  [70, 48, 5, 5, 4],
  [4, 44, 3, 3, 1],
  [23, 53, 1, 1, 2],
  [26, 93, 3, 3, 1.3],
  [46, 13, 4, 4, 2.3],
  [66, 34, 2, 2, 1.3],
  [36, 3, 3, 3, 3.3],
  [76, 54, 1, 1, 2.3],
  [56, 24, 3, 3, 1.3],
  [76, 25, 3, 3, 0.3],
  [56, 55, 1, 1, 2.31],
  [36, 35, 3, 3, 1.31],
  [86, 45, 2, 2, 2.31],
  [82, 85, 3, 3, 0.31],
  [81, 15, 1, 1, 1.31],
  [83, 35, 3, 3, 2.31],
  [23, 55, 2, 2, 4],
  [53, 50, 1, 1, 1.31],
  [83, 20, 1, 1, 1.51],
];

// top, left, width, height, opacity
export const MOON_CRATERS = [
  [23, 33, 17, 17, 0.5],
  [48, 10, 30, 30, 0.7],
  [90, 14, 20, 20, 0.4],
  [18, 63, 25, 25, 0.6],
  [33, 105, 10, 10, 0.8],
  [82, 55, 14, 14, 0.9],
  [122, 34, 20, 20, 0.6],
  [102, 84, 16, 16, 0.8],
  [63, 113, 20, 20, 0.9],
  [116, 119, 12, 12, 0.6],
];

// top, left, width, height, animationDuration, animationDelay
export const MOON_GLOWS = [
  [20, 30, 260, 260, 3, 0],
  [-20, -10, 340, 340, 3, 0.5],
];

// bottom, left, width, height
export const HILLS = [
  [183, 120, 20, 239],
  [259, 229, 28, 255],
  [211, 334, 28, 246],
];
