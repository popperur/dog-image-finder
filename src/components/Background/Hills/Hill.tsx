import { HillHolder } from "components/Background/Hills/styles.tsx";

interface HillProps {
  bottom: number;
  left: number;
  width: number;
  height: number;
}

function Hill({ bottom, left, width, height }: HillProps) {
  const stylesArray = [
    `bottom: ${bottom}px`,
    `left: ${left}px`,
    `width: ${width}px`,
    `height: ${height}px`,
  ];

  return <HillHolder $styles={stylesArray} />;
}

export default Hill;
