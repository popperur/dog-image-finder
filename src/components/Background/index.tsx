import {
  BackgroundBlock,
  BackgroundContainer,
  Ground,
} from "components/Background/styles.tsx";
import DogHouse from "components/Background/DogHouse.tsx";
import ShootingStars from "components/Background/ShootingStars.tsx";
import Dog from "components/Background/Dog";
import Moon from "components/Background/Moon";
import Stars from "components/Background/Stars";
import Hills from "components/Background/Hills";

// The animated background fetched from
// https://codepen.io/drexqq/pen/vvryLb/
// and reworked to have some randomized animations with React and styled components.
// Original version: LeeSungHee <3

function Background() {
  return (
    <BackgroundContainer>
      <BackgroundBlock>
        <Stars />
        <ShootingStars />
        <Moon />
        <Dog />
        <Hills />
        <DogHouse />
        <Ground />
      </BackgroundBlock>
    </BackgroundContainer>
  );
}

export default Background;
