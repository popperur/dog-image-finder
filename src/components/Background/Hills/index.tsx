import {
  BackHill,
  HillContainer,
} from "components/Background/Hills/styles.tsx";
import Hill from "components/Background/Hills/Hill.tsx";
import { HILLS } from "components/Background/constants.ts";

function Hills() {
  const hillsProps = HILLS.map(([bottom, left, width, height]) => ({
    bottom,
    left,
    width,
    height,
  }));

  return (
    <HillContainer>
      {hillsProps.map((hillProps, index) => (
        <Hill key={index} {...hillProps} />
      ))}
      <BackHill />
    </HillContainer>
  );
}

export default Hills;
