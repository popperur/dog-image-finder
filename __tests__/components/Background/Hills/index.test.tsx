import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Hills from "components/Background/Hills";
import { HILLS } from "components/Background/constants";

describe("Hills component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Hills />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows all hills and the backhill", () => {
    const { container } = render(<Hills />);

    expect(container.firstChild?.childNodes).toHaveLength(
      HILLS.length + 1 /* backhill */,
    );
  });
});
