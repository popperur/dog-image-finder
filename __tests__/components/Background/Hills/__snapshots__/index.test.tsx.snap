// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hills component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 183px;
  left: 120px;
  width: 20px;
  height: 239px;
}

.c1 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 259px;
  left: 229px;
  width: 28px;
  height: 255px;
}

.c2 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 211px;
  left: 334px;
  width: 28px;
  height: 246px;
}

.c3 {
  position: absolute;
  bottom: 0;
  right: 5%;
  height: 250px;
  width: 400px;
  background-color: #1f2731;
  border-top-left-radius: 500px;
  border-top-right-radius: 500px;
}

<div
    class=""
  >
    <div
      class="c0"
    />
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
  </div>
</DocumentFragment>
`;
